src/index.ts

```diff
@@ -65,38 +105,113 @@ export async function run(): Promise<void> {
     await execute(podman, args);
 
     core.info(`Successfully pushed ${imageToPush} to ${registryPath}.`);
-    core.setOutput('registry-path', registryPath);
+    core.setOutput("registry-path", registryPath);
+
+    // remove the pulled image from the Podman local registry
+    if (isPushingDockerImage) {
+        core.info(`Removing ${imageToPush} from the Podman local registry`);
+        await execute(podman, [ "rmi", imageToPush ]);
+    }
 
     try {
         const digest = (await fs.promises.readFile(digestFile)).toString();
         core.info(digest);
-        core.setOutput('digest', digest);
+        core.setOutput("digest", digest);
     }
     catch (err) {
         core.warning(`Failed to read digest file "${digestFile}": ${err}`);
     }
 }
 
-async function execute(executable: string, args: string[], execOptions: exec.ExecOptions = {}): Promise<{ exitCode: number, stdout: string, stderr: string }> {
+async function pullImageFromDocker(
+    imageName: string,
+    podman: string,
```

Comment:
it's not immediately clear what "podman" is here, can you name it `podmanPath` or something?

or, since the podman path is used throughout this program, you could make it globally available by creating a new function `getPodmanPath` which resolves the podman path, caches it (since the path won't change over the course of the action's lifetime) and returns it.

but, you can also keep it simple and just pass it around with a better name.